GCP: Add idle shutdown for Terraform deployed WS

Edwin-Pau opened the pull request:
All workstations deployed by Terraform now has auto idle shutdown enabled
by default. The default settings for idle shutdown is <enabled>, idle timer
at <240> minutes, CPU idle threshold at <20>%, and CPU polling interval
at <15> minutes. These default values were used to be in line with the idle 
shutdown settings for workstations deployed by CAM.

Commit:
1/2 GCP: add idle shutdown Terraform deployed WS

All workstations deployed by Terraform now has auto idle shutdown
enabled. The default settings for idle shutdown is <enabled>, idle timer
at <240> minutes, CPU idle threshold at <20>%, and CPU polling interval
at <15> minutes. These default values were used to be in line with the
idle shutdown settings for workstations deployed by CAM.

Signed-off-by: Edwin Pau <epau@teradici.com>

Commit:
2/2 GCP: add CPU polling interval variable

Added a CPU polling interval (in minutes) variable that will allow
users to change the default value from 15 minutes. By lowering the
CPU utilization polling interval, it will allow users to set a more
aggressive idle timer.

Signed-off-by: Edwin Pau <epau@teradici.com>

Edwin-Pau commented on modules/gcp/centos-gfx/centos-gfx-startup.sh.tmpl:
```diff
@@ -13,11 +13,36 @@ DECRYPT_URI="https://cloudkms.googleapis.com/v1/${kms_cryptokey_id}:decrypt"
+    then
+        return 1
+    fi
+}
```
Some of the commands to install idle timer needed retry logic as it sometimes failed. I will be creating a separate commit in the future to refactor any retry logic which will reuse this function.

Edwin-Pau commented on modules/gcp/centos-std/centos-std-startup.sh.tmpl:
```diff
@@ -13,11 +13,36 @@ DECRYPT_URI="https://cloudkms.googleapis.com/v1/${kms_cryptokey_id}:decrypt"
+    then
+        return 1
+    fi
+}
```
Same thing, see comment above.

Edwin-Pau commented on modules/gcp/centos-std/centos-std-startup.sh.tmpl:
```diff
@@ -92,6 +117,35 @@ install_pcoip_agent() {
+
+    if [[ $CPU_POLLING_INTERVAL -ne 15 ]]; then
+        log "--> Setting CPU polling interval to $${CPU_POLLING_INTERVAL} minutes"
+        sed -i "s/OnUnitActiveSec=15min/OnUnitActiveSec=$${CPU_POLLING_INTERVAL}min/g" /etc/systemd/system/CAMIdleShutdown.timer.d/CAMIdleShutdown.conf
```
The way to set CPU polling for Linux is different than Windows, I consulted Charanpreet on this. When changed to 1 min, the idle timer is much shorter and works as expected (instead of 15 + 15 mins for example).

shermanyin commented on modules/gcp/centos-gfx/centos-gfx-startup.sh.tmpl:
```diff
@@ -13,11 +13,36 @@ DECRYPT_URI="https://cloudkms.googleapis.com/v1/${kms_cryptokey_id}:decrypt"
+
+            log "--> Failed to run command. $${@}"
+            log "--> Retries left... $(( $max_retries - $retries ))"
+            ((n++))
```
what's n?

shermanyin commented on modules/gcp/win-gfx/win-gfx-startup.ps1.tmpl:
```diff
@@ -218,6 +222,39 @@ function PCoIP-Agent-Register {
+
+    $serviceName = "CAMIdleShutdown"
+    $path = "C:\Program Files\Teradici\PCoIP Agent\bin"
+    $is64 = $true
```
is this not needed or did you forget to use it somewhere else?

Edwin-Pau replied:
is64 not needed since we are always running on 64bit agent. Removed.

shermanyin commented on modules/gcp/win-gfx/win-gfx-startup.ps1.tmpl:
```diff
@@ -21,6 +21,10 @@ $DATA.Add("pcoip_registration_code", "${pcoip_registration_code}")
 $DATA.Add("admin_password", "${admin_password}")
 $DATA.Add("ad_service_account_password", "${ad_service_account_password}")
 
+$ENABLE_AUTO_SHUTDOWN     = "${enable_workstation_idle_shutdown}"
```
not familiar enough to know if "true" or "True" is what you want - you want a boolean so would it be
$ENABLE_AUTO_SHUTDOWN = $${enable_workstation_idle_shutdown}
so that after the template is rendered, it would become $true or $false?

Commit:
GCP: Fix and cleanup idle shutdown scripts

Fixed the variable used inside the retry function as part of the Centos
idle shutdown script.

Changed the enable idle shutdown variable used to a boolean instead
of a string for Windows workstation scripts. Removed unused variables
and added retry logic for disabling the idle shutdown service.

Signed-off-by: Edwin-Pau <epau@teradici.com>
